testsuite/tests/libratest/smoke_test.rs

```diff
@@ -160,13 +160,25 @@ impl TestEnvironment {
 fn setup_swarm_and_client_proxy(
     num_nodes: usize,
     client_port_index: usize,
+    template: Option<NodeConfig>,
```

Comment:
is there any other use case for this parameter?
if it's just to test small chunk size, I'd actually make it default behavior, and keep previous function signature

Reply:
Yep right now that's the only use case. I wonder in the future people might want to make custom environments to run tests on though?

Reply:
I'd prefer not to optimize for unknowns. We can add it later. Making this parameter as default actually will automatically cover all tests when StateSynchronizer is involved

Reply:
There's a good use case above where we can use this, right now we're internally setting weird defaults for smoke tests that aren't there for production, namely the ability to publish.

Reply:
hm, not sure why that specific default is weird though?
it's just ability to execute custom script. Doesn't have any implications on other tests and perf in general
would love to hear about more cases coming. Cause so far it looks like we could automatically cover multi-chunks in all related tests(e.g. restartability, full node sync, etc.) and all other call sites that keep passing None look weird

in short: a) I'd at least make this setting as default for all integration tests 
and b) would postpone adding complex signatures until there's real case. But don't have strong opinions on b), specially if you have more use cases in mind

Reply:
sort of unrelated but out of curiosity, do we want to only test multi-chunk sync in all cases? assuming we'd want to test single chunk as well, although that would probably happen less in prod

Reply:
I'd say always multi-chunk. It should cover richer set of cases

Reply:
weird was a poor choice of words, I should just say that they are configuring non-standard values. We ought to have smoke tests executing the default config unless there are reasons not to. You're right that this is isolated at this point in time, but I've seen the template all over the place and have not really seen any use cases except for this one. In light of the fact that we'd be moving to a default chunk of 2, I'm fine with making that the default within smoke test. So we can still benefit from @sunmilee's work at the interface between smoketests and libraswarm.

Reply:
Actually VMPublishingOption needs to be `VMPublishingOption::Open` just for `test_execute_custom_module_and_script` and doesn't need to be set as this for others. Should we make that as default too then?

Reply:
@davidiw: sorry for confusion. I have nothing against interface between smoketest and libraswarm. I just think signature of setup_swarm_and_client_proxy doesn't have to include template parameter yet. Cause IMO chunk size should be set globally and in all other places we just pass None for now. I'd add it once there's real need for it

Reply:
I'll keep the nodeconfig vs filepath refactor and make smoke test use the default chunk size instead of passing optional param.